enterprise/api/v1/views.py

```diff
@@ -1483,3 +1483,42 @@ def update(self, request, *args, **kwargs):
     )
     def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
+
+class EnterpriseUserLinkView(EnterpriseReadWriteModelViewSet):
+    """
+    View for
+    /enterprise/api/enterprise_user_link/{enterprise_customer_key}
+
+    Given a enterprise_customer_key, link user to the appropriate enterprise.
+
+    If the key is not found, returns 400
+    If the ket is not valid, return 422
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+
+    @action(methods=['post'], detail=False)
+    def link_user(self, request, enterprise_customer_key):
+        """
+        Post
+        Links user using enterprise_customer_key
+        """
+        try:
+            enterprise_customer_key_match = models.EnterpriseCustomerInviteKey.objects.get(pk=enterprise_customer_key)
+
+            if not enterprise_customer_key_match.is_valid:
+                return Response(
+                    {"detail": "Enterprise customer invite key is not valid"},
+                    status=status.HTTP_422_UNPROCESSABLE_ENTITY
+                )
+
+            enterprise_customer = models.EnterpriseCustomer.objects.get(
+                pk=enterprise_customer_key_match.enterprise_customer.uuid
+            )
+            models.EnterpriseCustomerUser.objects.get_or_create(
+                user_id=request.user.pk,
+                enterprise_customer=enterprise_customer,
+            )
```

Comment:
I think we need to create the enterprise learner role for the EnterpriseCustomerUser before we return the response. Might need to be a get_or_create? In case this endpoint gets called twice for the same enterprise we only have the 1 role 🤔

Reply:
So it looks like we have a [signal that does this](https://github.com/edx/edx-enterprise/blob/140c8b61c0400baa8b1f008382f389cc48740cac/enterprise/signals.py#L142) for us! I tested this in my local env and when enterprise learners are created their roles are automatically assigned.

Reply:
iiiinteresting. good find!